src/NzbDrone.Core/Indexers/Definitions/Redacted.cs

```diff
@@ -41,13 +74,286 @@ protected override IndexerCapabilities SetCapabilities()
+            caps.Categories.AddCategoryMapping(3, NewznabStandardCategory.BooksEBook, "E-Books");
             caps.Categories.AddCategoryMapping(4, NewznabStandardCategory.AudioAudiobook, "Audiobooks");
-            caps.Categories.AddCategoryMapping(5, NewznabStandardCategory.Movies, "E-Learning Videos");
-            caps.Categories.AddCategoryMapping(6, NewznabStandardCategory.TV, "Comedy");
-            caps.Categories.AddCategoryMapping(7, NewznabStandardCategory.Books, "Comics");
+            caps.Categories.AddCategoryMapping(5, NewznabStandardCategory.MoviesOther, "E-Learning Videos");
+            caps.Categories.AddCategoryMapping(6, NewznabStandardCategory.TVOther, "Comedy");
+            caps.Categories.AddCategoryMapping(7, NewznabStandardCategory.BooksComics, "Comics");
 
             return caps;
         }
     }
+
+    public class RedactedRequestGenerator : IIndexerRequestGenerator
+    {
+        public RedactedSettings Settings { get; set; }
+        public IndexerCapabilities Capabilities { get; set; }
+        public string BaseUrl { get; set; }
+        public Func<IDictionary<string, string>> GetCookies { get; set; }
+        public Action<IDictionary<string, string>, DateTime?> CookiesUpdater { get; set; }
+        public IHttpClient HttpClient { get; set; }
+
+        public RedactedRequestGenerator()
+        {
+        }
+
+        public IndexerPageableRequestChain GetSearchRequests(MusicSearchCriteria searchCriteria)
+        {
+            var pageableRequests = new IndexerPageableRequestChain();
+
+            pageableRequests.Add(GetRequest(string.Format("&artistname={0}&groupname={1}", searchCriteria.Artist, searchCriteria.Album)));
+
+            return pageableRequests;
+        }
+
+        public IndexerPageableRequestChain GetSearchRequests(BookSearchCriteria searchCriteria)
+        {
+            var pageableRequests = new IndexerPageableRequestChain();
+
+            pageableRequests.Add(GetRequest(searchCriteria.SanitizedSearchTerm));
+
+            return pageableRequests;
+        }
+
+        public IndexerPageableRequestChain GetSearchRequests(MovieSearchCriteria searchCriteria)
+        {
+            return new IndexerPageableRequestChain();
+        }
+
+        public IndexerPageableRequestChain GetSearchRequests(TvSearchCriteria searchCriteria)
+        {
+            return new IndexerPageableRequestChain();
+        }
+
+        public IndexerPageableRequestChain GetSearchRequests(BasicSearchCriteria searchCriteria)
+        {
+            var pageableRequests = new IndexerPageableRequestChain();
+
+            pageableRequests.Add(GetRequest(searchCriteria.SanitizedSearchTerm));
+
+            return pageableRequests;
+        }
+
+        private IEnumerable<IndexerRequest> GetRequest(string searchParameters)
+        {
+            // Retrieve passkey if not already done
+            if (string.IsNullOrWhiteSpace(Settings.Passkey))
```

Comment:
Does this passkey ever expire? if so there is no way for it to be re-generated the way the code is written.

Reply:
Since this is the same passkey used in the torrent files themselves, no, this should never expire.  However, a user can manually go into their RED profile and manually reset the passkey.  But, that actually shouldn't cause any problems here.  As far as I can tell, the passkey is never committed to the database, and instead, persists only for this particular instance of  RedactedSettings object.  In practice, this seems to fetch the passkey once per query.  But the passkey shouldn't get fetched for each page of results, only once.

This definitely feels a bit hacky though, so I am very open to suggestions on better ways to manage the passkey.

Reply:
Yea, I'd prefer to keep it automated, the less we rely on the user the better

Reply:
i dont see how this is any different than having to end the api key for united based trackers like blutopia and even beyondhd. beyondhd you have to type in the api key and rss key. id rather have to add the api key like any other tracker. its been this way for a long time why change to this automated way that may be more of a headache to get working again if a user deletes their api key or regenerates it?

Reply:
I removed the `if (string.IsNullOrWhiteSpace(Settings.Passkey))` check, this way it is both automated and will continue to work even if the user generates a new passkey.